src/main/java/org/isf/exa/service/ExamIoOperations.java

```diff
@@ -91,19 +92,67 @@ public List<ExamType> getExamType() throws OHServiceException {
 	}
 
 	/**
-	 * Insert a new {@link Exam} in the DB.
-	 * 
+	 * Insert a new {@link Exam} with exam rows.
+	 * @param payload - the {@link Exam} to insert
+	 * @param rows - the {@link List<String>} to associate as exam rows
+	 * @return the newly persisted {@link Exam}.
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public Exam create(Exam payload, List<String> rows) throws OHServiceException {
+		Exam exam = repository.save(payload);
+		if (exam.getProcedure() == 3) {
+			return exam;
+		}
+		if (rows != null && !rows.isEmpty()) {
+			rowRepository.saveAll(rows.stream().map(description -> new ExamRow(exam, description)).toList());
+		}
+		return exam;
+	}
+
+	/**
+	 * Update an existing {@link Exam} with exam rows.
+	 * @param payload - the {@link Exam} to insert
+	 * @param rows - the {@link List<String>} to associate as exam rows
+	 * @return the newly persisted {@link Exam}.
+	 * @throws OHServiceException
+	 */
+	@Transactional
+	public Exam update(Exam payload, List<String> rows) throws OHServiceException {
+		Exam oldExam = findByCode(payload.getCode());
+		Exam exam = repository.save(payload);
+		List<ExamRow> examRows = rowRepository.findAllByExam_CodeOrderByDescription(exam.getCode());
+		if (exam.getProcedure() == 3) {
+			if (oldExam.getProcedure() != 3) {
+				rowRepository.deleteAll(examRows);
```

Comment:
Maybe the two if could be combined:

```suggestion
		if (exam.getProcedure() == 3 && oldExam.getProcedure() != 3) {
				rowRepository.deleteAll(examRows);
```

Reply:
`
if (exam.getProcedure() == 3) {
	if (oldExam.getProcedure() != 3) {
		rowRepository.deleteAll(examRows);
	}
	return exam;
}
`

I want to just return the saved exam when its procedure is 3(no extra comptation). I also want to do delete exam rows if old exam procedure was not 3.

Doing like your're saying doesn't solve that.
Am I wrong ?

Reply:
so you could put the whole extra computation in a else block and having only a return (more readable)

Reply:
Oh, I see, that works !